docs/source/api/references/graphstorm.model.rst

```diff
@@ -3,55 +3,98 @@
 graphstorm.model
 ===================
 
-    A GraphStorm model may contain three components:
+    GraphStorm provides a set of Graph Neural Network (GNN) modules. By combining them
+    in proper ways, users can build various GNN models for different tasks.
 
-    * Input layer: a set of modules to convert input data for different use cases,
-      e.g., embedding texture features.
-    * Encoder: a set of Graph Neural Network modules 
-    * Decoder: a set of modules to convert results from encoders for different tasks,
-      e.g., classification, regression, or link prediction.
+    A GNN model in GraphStorm normally contains four components:
 
-    Currently GraphStorm releases the first two set of components.
+    * Input layer: an input encoder that converts input node/edge features into embeddings
+      with the given hidden dimensions. The output of an input layer will become the input of
+      the GNN layer, or the decoder layer if GNN is not needed.
+    * GNN layer (Optional): a GNN encoder that performs the message passing computation. 
+      The outputs of a GNN layer are embeddings of nodes that wil be used in the decoder layer.
+    * Decoder layer: a task specific decoder that converts results from either a GNN layer or 
+      an input layer into loss values for different GML tasks, e.g., classification, regression,
+      or link prediction.
+    * Model optimizer: GraphStorm model classes have a built-in model optimizer, which should
+      be initialized during GraphStorm GNN model object construction.
 
-    If users would like to implement their own model, the best practice is to extend the corresponding ``***ModelBase``, and implement the abstract methods.
+    If users would like to implement their own GNN models, a suggested practice is to extend a
+    base GNN model class and its corresponding interface, e.g., ``GSgnnNodeModelBase`` and
+    ``GSgnnNodeModelInterface``, and implement the required abstract methods.
+
+    If users just want to build their own message passing methods, a suggested practice is to
+    create their own GNN encoders by extending the ``GraphConvEncoder`` base class, and
+    implement the ``forward(self, blocks, h)`` function, which will be called by GraphStorm GNN
+    model classes within their own ``forward()`` function.
```

Comment:
I think these two forward functions should refer the same one, right? So we can say ``and
    implement the `forward(self, blocks, h)` function, which will be called by GraphStorm GNN
    model classes.``

Reply:
No, the two `forward()`s are different. GNN model's `forward()` function normally takes different arguments because model's input is for the `InputLayer`, while these `Encoder`s ask for MFGs and embeddings.